app/src/lib/stores/repositories-store.ts

```diff
@@ -485,4 +530,43 @@ export class RepositoriesStore extends BaseStore {
 
     return record!.lastPruneDate
   }
+
+  public async isBranchProtected(
+    gitHubRepository: GitHubRepository,
+    branchName: string
+  ): Promise<boolean> {
+    if (gitHubRepository.dbID === null) {
+      return fatalError(
+        'unable to get protected branches, GitHub repository has a null dbID'
+      )
+    }
+
+    const key = getKey(gitHubRepository.dbID, branchName)
+
+    const existing = this.branchProtectionCache.get(key)
+    if (existing !== undefined) {
+      return existing
+    }
+
+    const result = await this.db.protectedBranches.get({
```

Comment:
So the problem with this approach of using an object instead of an array is that it'll be the [exact same thing as doing a `.where().equals().first()`](https://github.com/dfahlander/Dexie.js/blob/v2.0.0/src/Dexie.js#L958-L959). I.e. it will not be a primary key lookup. I bet this is just another case of the Dexie typings not being perfect. Did you try casting the array to `Array<any>`?

Reply:
The problem here was the original database declaration wasn't updated from `Key` being `number` to being `[number, string]` to match the new constraints. Fixed in c64f057551ba0eb9327f3c2b5bd26f280abb251b.